### ISSUE TITLE
Issue with using "As-User" Header and Creating Metadata

### ISSUE THREAD

**Issue by aowolabi** [box/box-python-sdk#252]
The Metadata object appears to be the only class to explicitly set headers (see: [`create`](https://github.com/box/box-python-sdk/blob/v2.0.0a11/boxsdk/object/metadata.py#L206) and [`update`](https://github.com/box/box-python-sdk/blob/v2.0.0a11/boxsdk/object/metadata.py#L162)). This causes any default headers that may have been set to be [ignored](https://github.com/box/box-python-sdk/blob/v2.0.0a11/boxsdk/session/box_session.py#L398). If you happen to be relying on `as_user` impersonation functionality, a `404` resource not found response is received for the file that metadata is being added to. Is this intended behavior?

**Comment by Jeff-Meadows** [comment]
This isn't intended - thanks for the report!

**Comment by aowolabi** [comment]
Ah, good to know... should be a simple enough fix. What are your thoughts, something like: 

```
headers = self._session._default_headers.copy()
headers.update({b'Content-Type': b'application/json'})
return self._session.post(
            self.get_url(),
            data=json.dumps(metadata),
            headers=headers,
        ).json()
```

**Comment by mattwiller** [comment]
This is fixed on `master` and will be released with v2.0.0
